File: backend/src/main/java/com/kiwi/features/goals/data/GoalDTO.java
package com.kiwi.features.goals.data;

import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode
@ToString
public class GoalDTO {
    private Long id;
    private String name;
    private String action;
    private Integer target;
    private String type;
    private String category;
    private Integer reward;
    private String onCompletedAction;
    private String onCompletedEntity;
    private Integer onCompletedEntityId;
    private String onCompletedEvent;
}

File: mobile/app/src/main/java/com/bellako/kiwi/features/goals/data/GoalDTO.kt
package com.bellako.kiwi.features.goals.data

data class GoalDTO(
    val id: Long,
    val name: String,
    val target: Int,
    val action: String,
    val type: String,
    val category: String,
    val reward: Int,
    val onCompletedEvent: String = "_",
    val onCompletedEntityId: Int = 0,
)
